src/libmongoc/src/mongoc/mongoc-scram.c

```diff
@@ -1150,4 +1229,176 @@ _mongoc_sasl_prep (const char *in_utf8, int in_utf8_len, bson_error_t *err)
    return bson_strdup (in_utf8);
 #endif
 }
+
+int
+_mongoc_utf8_char_length (const char *c)
+{
+   int length;
+
+   // UTF-8 characters are either 1, 2, 3, or 4 bytes and the character length
+   // can be determined by the first byte
+   if ((*c & 0x80) == 0)
```

Comment:
```suggestion
   const uint8_t unit = (uint8_t) *c;

   // UTF-8 characters are either 1, 2, 3, or 4 bytes and the character length
   // can be determined by the first byte
   if ((unit & UINT8_C (0x80)) == 0) {
      return 1u;
   }
```
Use `uint8_t` to represent code units. Cast `char -> uint8_t` before operating on code units, and use `UINT8_C ()` to ensure correct and consistent suffixes are used for integer literals.